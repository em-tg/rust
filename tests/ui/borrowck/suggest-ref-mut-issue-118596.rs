//@ check-pass
fn main() {
    let y = Some(0);
    if let Some(x) = y {
        x = 2; //~ WARNING cannot assign twice to immutable variable `x`
    }

    let mut arr = [1, 2, 3];
    let [x, ref xs_hold @ ..] = arr;
    x = 0; //~ WARNING cannot assign twice to immutable variable `x`
    eprintln!("{:?}", arr);
}
