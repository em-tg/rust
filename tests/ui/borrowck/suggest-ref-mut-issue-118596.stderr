warning[E0384]: cannot assign twice to immutable variable `x`
  --> $DIR/suggest-ref-mut-issue-118596.rs:5:9
   |
LL |     if let Some(x) = y {
   |                 - first assignment to `x`
LL |         x = 2;
   |         ^^^^^ cannot assign twice to immutable variable
   |
   = note: `#[warn(mut_non_mut)]` on by default
help: consider making this binding mutable
   |
LL |     if let Some(mut x) = y {
   |                 +++
help: to modify the original value, take a borrow instead
   |
LL |     if let Some(ref mut x) = y {
   |                 +++++++

warning[E0384]: cannot assign twice to immutable variable `x`
  --> $DIR/suggest-ref-mut-issue-118596.rs:10:5
   |
LL |     let [x, ref xs_hold @ ..] = arr;
   |          - first assignment to `x`
LL |     x = 0;
   |     ^^^^^ cannot assign twice to immutable variable
   |
help: consider making this binding mutable
   |
LL |     let [mut x, ref xs_hold @ ..] = arr;
   |          +++
help: to modify the original value, take a borrow instead
   |
LL |     let [ref mut x, ref xs_hold @ ..] = arr;
   |          +++++++

warning: 2 warnings emitted

For more information about this error, try `rustc --explain E0384`.
